### ISSUE TITLE
struct2depth inference.py imagenet norm not applied

### ISSUE THREAD

**Issue by lkskstlr** [tensorflow/models#7343]
### System information
- **What is the top-level directory of the model you are using**: models/research/struct2depth
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**: yes
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Linux Ubuntu 16.04
- **TensorFlow installed from (source or binary)**: Binary
- **TensorFlow version (use command below)**: v1.12.0-0-ga6d8ffae09 1.12.0
- **Bazel version (if compiling from source)**:
- **CUDA/cuDNN version**: 9.0.176/7.6.0.64
- **GPU model and memory**: TITAN Xp 12189MiB
- **Exact command to reproduce**: `python inference.py --model_ckpt /root/data/struct2depth/ckpt/model-100000 --output_dir /root/data/struct2depth/eval-result --input_dir /root/data/struct2depth/eval-data/`

### Describe the problem
I think that when running inference imagenet normalization is not applied even when the flag is set. Also described in a pull request: https://github.com/tensorflow/models/pull/7342 . @aneliaangelova could you maybe take a look please? Sorry if the error is on my side.

If this is really a bug, I would like to hear your thoughts on how this influences the results :)

### Source code / logs
I have inserted two modifications to track the issue:

In `inference.py` I added
```
logging.info("min = %s, max = %s", np.amin(im_batch), np.amax(im_batch))
```
in line 204, i.e. before calling `est_depth = inference_model.inference_depth(im_batch, sess)`.

In `model.py` I added
```
logging.info("input_image 1 %s", input_image)
if self.imagenet_norm:
    input_image = (input_image - reader.IMAGENET_MEAN) / reader.IMAGENET_SD
logging.info("input_image 2 %s", input_image)

print_op1 = tf.print("min(input_image): ", tf.reduce_min(input_image), "\n", output_stream=sys.stderr)
print_op2 = tf.print("max(input_image): ", tf.reduce_max(input_image), "\n", output_stream=sys.stderr)

with tf.control_dependencies([print_op1, print_op2]):
    est_disp, _ = nets.disp_net(architecture=self.architecture,
                                image=input_image,
                                use_skip=self.use_skip,
                                weight_reg=self.weight_reg,
                                is_training=True)
```
in `def build_depth_test_graph(self)` and
```
logging.info("self.input_image %s", self.input_image)
```
in `def inference_depth(self, inputs, sess)`.

This gives (I excluded other lines) the following output:
```
input_image 1 Tensor("depth_prediction/raw_input:0", shape=(1, 128, 416, 3), dtype=float32)
input_image 2 Tensor("depth_prediction/truediv:0", shape=(1, 128, 416, 3), dtype=float32)
...
min = 0.003921569, max = 1.0
self.input_image Tensor("depth_prediction/truediv:0", shape=(1, 128, 416, 3), dtype=float32)
max(input_image):  1 
min(input_image):  0.00392156886 
```

Using the modification given in the pull request the output for min/max becomes
```
min = 0.003921569, max = 1.0
max(input_image):  2.64 
min(input_image):  -2.10077929 
```

Thanks for your help
-Lukas

**Comment by aneliaangelova** [comment]
I don't believe this is a bug as the reader calls tf.image.convert_image_dtype
 which implicitly does normalization. This is a bit tricky as it appears as though the imagenet flag is the one to control it but I believe the code is working as intended. 
No additional normalization is needed. Thanks!

**Comment by kumariko** [comment]
Please refer https://github.com/tensorflow/models/issues/7343#issuecomment-521834654 and let us know if you still need help on this issue? Thanks
